src/utils/ao.ts

```diff
@@ -171,22 +172,86 @@ export async function spawnANT({
     }
   }
 
-  // for hyperbeam caching, due to a SU issue, we need to send a second message to the ANT to cache the state
+  // Note: for hyperbeam caching, due to a SU issue, we need to send a second message to the ANT to cache the state
   // We wait for the first message to be processed before sending the second one to ensure this is the second message
-  const processApi = new AOProcess({
-    processId,
-    ao,
-    logger,
-  });
-  await processApi.send({
-    tags: [{ name: 'Action', value: 'State' }],
-    signer,
-  });
+  // We use the resulting state to check the owner of the ANT is set in the registry. Should this be patched on MUs,
+  // we can convert to just a simple dry-run to avoid sending/signing another message.
+  let owner: WalletAddress | undefined;
+  try {
+    const processApi = new AOProcess({
+      processId,
+      ao,
+      logger,
+    });
+    const { id } = await processApi.send({
+      tags: [{ name: 'Action', value: 'State' }],
+      signer,
+    });
+    const stateResult = await ao.result({
+      process: processId,
+      message: id,
+    });
+    if (stateResult === undefined) {
+      throw new Error('Failed to get state result');
+    }
+    const { Owner } = JSON.parse(stateResult.Messages?.[0]?.Data ?? '{}') as {
+      Owner: WalletAddress;
+    };
+    owner = Owner;
```

Comment:
Owner should default in the params to spawnANT, and this path should only run if owner is undefined.

... alternatively, _always_ run it but throw if the passed in owner doesn't match the _actual_ owner that we get back here to further validate the spawn?

Reply:
Also, why not use ANT.getState?

Reply:
discussed